### PR TITLE
update alloy

### PR DESCRIPTION
**Pull request by supernovahs**
this pr updates alloy to latest, uses forge bind --alloy to generate bindings  . Since those bindings use alloy::providers, alloy::primitives. I had to change the use of alloy_provider to alloy::providers etc for the whole codebase, else it gives the classic error  `Same variable , but a different version`

### CODE REVIEW

**Comment by supernovahs** [comment]
there are some unused binding files which i have not deleted. should i do it ?

**Comment by supernovahs** [comment]
docker was throwing ssh authentication, so changed the ssh url to http  for submodule

File: crates/chainio/clients/avsregistry/src/fake_reader.rs [hunk 1]
```diff
@@ -45,7 +43,7 @@ impl AvsRegistryReader for FakeAvsRegistryReader {
         Ok(vec![vec![OperatorStateRetriever::Operator {
             operator: self.operator_address,
             operatorId: self.operator_id,
-            stake: OPERATOR_STAKE,
+            stake: U96::from(123),
```
Comment (pablodeymo) [inline crates/chainio/clients/avsregistry/src/fake_reader.rs:46]: why did you remove the constant?

Comment (supernovahs) [reply crates/chainio/clients/avsregistry/src/fake_reader.rs:46]: because  it required use of lazy constant and was used only once. shoudl i use lazy?

**Review by pablodeymo — APPROVED** [review]
